tests/whimsylib/say_test.py

```diff
@@ -0,0 +1,136 @@
+import io
+import re
+from unittest.mock import call
+from unittest.mock import patch
+import unittest
+
+from whimsylib.globals import G
+from whimsylib import say
+from tests import common
+
+
+def _to_generator(string):
+    for character in string:
+        yield character
+
+
+def _sifted_edits(original, augmented):
+    """Reconstructs original from augmented by removing characters.
+
+    If original is a disjoint substring of augmented (i.e., if a series of
+    addition edits/interpolations can get from original to augmented), this
+    method returns the tuple (original, edits).
+
+    Otherwise, this method will return the longest left substring that can be
+    so reconstructed and all edits encountered up to that point. In either case,
+    the returned values will be strings.
+
+    This is equivalent to assessing that the edit distance between the two
+    strings consists only in a series of additions (not removals or
+    substitutions).
+    """
+    reconstructed = []
+    edits = []
+
+    aug_gen = _to_generator(augmented)
+
+    for oc in original:
+        while True:
+            try:
+                next_aug = next(aug_gen)
+            except StopIteration:
+                break
+
+            if oc == next_aug:
+                reconstructed.append(next_aug)
```

Comment:
```suggestion
                reconstructed.append(next_aug)
                break
```

Reply:
Somehow, we lost that break statement, but tests aren't failing in cronenbroguelike.

This is what happens when you write a body horror game: it ends up haunted.